builtin/logical/pki/path_resign_crls.go

```diff
@@ -0,0 +1,274 @@
+			},
+			formatParam: {
+				Type: framework.TypeString,
+				Description: `The format of the combined CRL, can be "pem" or "der". If "der", the value will be
+base64 encoded. Defaults to "pem".`,
+				Default: "pem",
+			},
+		},
+		Operations: map[logical.Operation]framework.OperationHandler{
+			logical.UpdateOperation: &framework.PathOperation{
+				Callback: b.pathUpdateResignCrlsHandler,
+			},
+		},
+
+		HelpSynopsis: `Combine and sign with the provided issuer different CRLs `,
+		HelpDescription: `Provide two or more PEM encoded CRLs signed by the issuer, 
+normally from separate Vault clusters to be combined and signed.`,
+	}
+}
+
+func (b *backend) pathUpdateResignCrlsHandler(ctx context.Context, request *logical.Request, data *framework.FieldData) (*logical.Response, error) {
+	if b.useLegacyBundleCaStorage() {
+		return logical.ErrorResponse("This API cannot be used until the migration has completed"), nil
+	}
+
+	issuerRef := getIssuerRef(data)
+	crlNumber := data.Get(crlNumberParam).(int)
+	deltaCrlNumber := data.Get(deltaCrlNumberParam).(int)
+	nextUpdateStr := data.Get(nextUpdateParam).(string)
+	rawCrls := data.Get(crlsParam).([]string)
+
+	format, err := getCrlFormat(data.Get(formatParam).(string))
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	nextUpdateOffset, err := time.ParseDuration(nextUpdateStr)
+	if err != nil {
+		return logical.ErrorResponse("invalid value for %s: %v", nextUpdateParam, err), nil
+	}
+
+	if nextUpdateOffset <= 0 {
+		return logical.ErrorResponse("%s parameter must be greater than 0", nextUpdateParam), nil
+	}
+
+	if crlNumber < 0 {
+		return logical.ErrorResponse("%s parameter must be 0 or greater", crlNumberParam), nil
+	}
+	if deltaCrlNumber < 0 {
+		return logical.ErrorResponse("%s parameter must be 0 or greater", deltaCrlNumberParam), nil
+	}
+
+	if issuerRef == "" {
+		return logical.ErrorResponse("%s parameter cannot be blank", issuerRefParam), nil
+	}
+
+	providedCrls, err := decodePemCrls(rawCrls)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	sc := b.makeStorageContext(ctx, request.Storage)
+	issuerEntity, err := getIssuer(sc, issuerRef)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	if err := verifyCrlsAreFromIssuer(issuerEntity, providedCrls); err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	revokedCerts, warnings, err := getAllRevokedCerts(providedCrls)
+	if err != nil {
+		return logical.ErrorResponse(err.Error()), nil
+	}
+
+	now := time.Now()
+	template := &x509.RevocationList{
+		SignatureAlgorithm:  issuerEntity.caBundle.RevocationSigAlg,
+		RevokedCertificates: revokedCerts,
+		Number:              big.NewInt(int64(crlNumber)),
+		ThisUpdate:          now,
+		NextUpdate:          now.Add(nextUpdateOffset),
+	}
+
+	if deltaCrlNumber > 0 {
+		ext, err := certutil.CreateDeltaCRLIndicatorExt(int64(deltaCrlNumber))
```

Comment:
re: naming, it just makes this seem like it is doing the wrong thing, but in reality, it is doing the correct thing.

Also, FWIW, I think we do technically allow zero-valued CRLNumbers, so you could create a CRL with number 0, then create a delta with number 1, pointing back to value zero.

So maybe default to `-1` in the API?

Reply:
Yup agree with the name change and the -1 for this use case.

Reply:
Fixed.